### PR TITLE
[#8368] [Platform] [API] Cloud Provider controller error handling.

### PR DESCRIPTION
**Pull request by jitendra-12113**
Description:

Use the new global exception handling in the Cloud Provider controller.

Test Plan:

Its mostly refactor. Unit tests are updated.
Run server and test basic operations.

**SBT TEST O/p:**

[info] Test com.yugabyte.yw.commissioner.TaskGarbageCollectorTest.testStart_disabled started
[info] Test com.yugabyte.yw.commissioner.TaskGarbageCollectorTest.testPurge_noneStale started
[info] Test com.yugabyte.yw.commissioner.TaskGarbageCollectorTest.testIgnorePromError started
[info] Test com.yugabyte.yw.commissioner.TaskGarbageCollectorTest.testPurge_invalidData started
[info] Test run finished: 0 failed, 0 ignored, 7 total, 0.435s
[error] Failed: Total 1286, Failed 1, Errors 0, Passed 1281, Skipped 4
[error] Failed tests:
[error]         com.yugabyte.yw.controllers.ApiDiscoveryControllerTest
[error] (Test / test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 1165 s (19:25), completed 20 Apr, 2021 11:16:36 PM
hasher@LAP-LIN-908:~/custom/yugabyte-db/managed$

### CODE REVIEW

File: managed/src/main/java/com/yugabyte/yw/controllers/CloudProviderController.java [hunk 1]
```diff
@@ -213,7 +209,8 @@ public Result create(UUID customerUUID) throws IOException {
     } catch (RuntimeException e) {
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/controllers/CloudProviderController.java:209]: Fix RuntimeException catching business

**Review by sb-yb — CHANGES_REQUESTED** [review]

File: managed/src/main/java/com/yugabyte/yw/controllers/CloudProviderController.java (hunk 1 above)
Comment (jitendra-12113) [reply managed/src/main/java/com/yugabyte/yw/controllers/CloudProviderController.java:209]: Created another PR to resolve merge conflicts and this review comments 
https://github.com/yugabyte/yugabyte-db/pull/8472

Comment (jitendra-12113) [reply managed/src/main/java/com/yugabyte/yw/controllers/CloudProviderController.java:209]: Done.

new PR: https://github.com/yugabyte/yugabyte-db/pull/8472

**Comment by jitendra-12113** [comment]
Fix review comments and merge conflict as part of separate PR https://github.com/yugabyte/yugabyte-db/pull/8472

**Pull request closed without merging**